packages/backend/src/plugins/catalog.ts

```diff
@@ -14,37 +14,21 @@
  * limitations under the License.
  */
 
+import { useHotCleanup } from '@backstage/backend-common';
 import {
+  CatalogBuilder,
   createRouter,
-  DatabaseEntitiesCatalog,
-  DatabaseLocationsCatalog,
-  DatabaseManager,
-  HigherOrderOperations,
-  LocationReaders,
   runPeriodically,
 } from '@backstage/plugin-catalog-backend';
 import { PluginEnvironment } from '../types';
-import { useHotCleanup } from '@backstage/backend-common';
-
-export default async function createPlugin({
-  logger,
-  config,
-  reader,
-  database,
-}: PluginEnvironment) {
-  const locationReader = new LocationReaders({ logger, reader, config });
 
-  const db = await DatabaseManager.createDatabase(await database.getClient(), {
-    logger,
-  });
-  const entitiesCatalog = new DatabaseEntitiesCatalog(db);
-  const locationsCatalog = new DatabaseLocationsCatalog(db);
-  const higherOrderOperation = new HigherOrderOperations(
+export default async function createPlugin(env: PluginEnvironment) {
+  const builder = new CatalogBuilder(env);
+  const {
     entitiesCatalog,
     locationsCatalog,
-    locationReader,
-    logger,
-  );
+    higherOrderOperation,
```

Comment:
Could we throw in a rename of this too? It's like calling a function "higherOrderFunction" imo

Reply:
Nah not here :) I am planning to reorder things a bit so the entire thing may go away

Reply:
👍, that soon btw? Best get as many breaking changes as possible into the next release

Reply:
We just released yesterday! :) But yeah we'll see, it's hack week after all